staging/src/k8s.io/cloud-provider/fake/fake.go

```diff
@@ -321,13 +323,20 @@ func (f *Cloud) InstanceMetadata(ctx context.Context, node *v1.Node) (*cloudprov
 	f.addCall("instance-metadata-by-provider-id")
 	f.addressesMux.Lock()
 	defer f.addressesMux.Unlock()
+
+	providerID := node.Spec.ProviderID
+	_, ok := f.ExtID[types.NodeName(node.Name)]
```

Comment:
Realiziing now that similar to ExtID, `ProviderID` should probably be of type `map[types.NodeName]string`. And here we should check `f.ProviderID` instead of `f.ExtdID`.

Reply:
gotcha! Will update